commit a8be0acf2ac5eba5e7b238c4976732d312d1e714
Author: Jacobo Cabaleiro <obmun.h@gmail.com>
Date:   Sun Mar 22 00:41:29 2015 +0000

    Support boost graph lib > 1.48
    
    Boost > 1.48 changed the graph library in an uncompatible way with VTK
    5.8 code: http://www.paraview.org/Bug/view.php?id=12772
    
    Fedora included a patch in his 5.8 builds (vtk-boost-1.48.0-bfs.patch)
    but this patch is only valid for boost 1.48, as later versions once more
    changed the library iface in a non compatible way.
    
    These changes properly fix the issue for all boost versions. Commiting
    so I can generate a patch to integrate into the Fedora RPM build spec.

diff --git a/Infovis/vtkBoostBreadthFirstSearchTree.cxx b/Infovis/vtkBoostBreadthFirstSearchTree.cxx
index c789f6b..fc06056 100644
--- a/Infovis/vtkBoostBreadthFirstSearchTree.cxx
+++ b/Infovis/vtkBoostBreadthFirstSearchTree.cxx
@@ -47,6 +47,23 @@ using namespace boost;
 vtkStandardNewMacro(vtkBoostBreadthFirstSearchTree);
 
 
+#if BOOST_VERSION >= 104800 // Boost 1.48.x
+namespace {
+  vtkIdType unwrap_edge_id(vtkEdgeType const &e)
+  {
+    return e.Id;
+  }
+  vtkIdType unwrap_edge_id(boost::detail::reverse_graph_edge_descriptor<vtkEdgeType> const &e)
+  {
+# if BOOST_VERSION == 104800
+    return e.underlying_desc.Id;
+# else
+    return e.underlying_descx.Id;
+# endif
+  }
+}
+#endif
+
 // Redefine the bfs visitor, the only visitor we
 // are using is the tree_edge visitor.
 template <typename IdMap>
@@ -95,7 +112,12 @@ public:
 
     // Copy the vertex and edge data from the graph to the tree.
     tree->GetVertexData()->CopyData(graph->GetVertexData(), v, tree_v);
+#if BOOST_VERSION < 104800 // Boost 1.48.x
     tree->GetEdgeData()->CopyData(graph->GetEdgeData(), e.Id, tree_e.Id);
+#else
+    tree->GetEdgeData()->CopyData(graph->GetEdgeData(),
+				  unwrap_edge_id(e), tree_e.Id);
+#endif
   }
 
 private:
